pkg/doc/jose/jwk_test.go

```diff
@@ -0,0 +1,238 @@
+							"x": "sEHL6KXs8bUz9Ss2qSWWjhhRMHVjrog0lzFENM132R8",
+							"alg": "EdDSA"
+						}`,
+			},
+			{
+				name: "get public key bytes RSA JWK",
+				jwkJSON: `{
+							"kty": "RSA",
+							"e": "AQAB",
+							"use": "enc",
+							"kid": "sample@sample.id",
+							"alg": "RS256",
+							"n": "1hOl09BUnwY7jFBqoZKa4XDmIuc0YFb4y_5ThiHhLRW68aNG5Vo23n3ugND2GK3PsguZqJ_HrWCGVuVlKTmFg` +
+					`JWQD9ZnVcYqScgHpQRhxMBi86PIvXR01D_PWXZZjvTRakpvQxUT5bVBdWnaBHQoxDBt0YIVi5a7x-gXB1aDlts4RTMpfS9BPmEjX` +
+					`4lciozwS6Ow_wTO3C2YGa_Our0ptIxr-x_3sMbPCN8Fe_iaBDezeDAm39xCNjFa1E735ipXA4eUW_6SzFJ5-bM2UKba2WE6xUaEa5G1` +
+					`MDDHCG5LKKd6Mhy7SSAzPOR2FTKYj89ch2asCPlbjHTu8jS6Iy8"
+						}`,
+			},
+			{
+				name: "get public key bytes EC P-526 JWK",
+				jwkJSON: `{
+							"kty": "EC",
+							"use": "enc",
+							"crv": "P-256",
+							"kid": "sample@sample.id",
+							"x": "JR7nhI47w7bxrNkp7Xt1nbmozNn-RB2Q-PWi7KHT8J0",
+							"y": "iXmKtH0caOgB1vV0CQwinwK999qdDvrssKhdbiAz9OI",
+							"alg": "ES256"
+						}`,
+			},
+			{
+				name: "get public key bytes EC SECP256K1 JWK",
+				jwkJSON: `{
+							"kty": "EC",
+    						"use": "enc",
+    						"crv": "secp256k1",
+    						"kid": "sample@sample.id",
+    						"x": "wQehEGTVCu32yp8IwTaBCqPUIYslyd-WoFRsfDKE9II",
+    						"y": "rIJO8RmkExUecJ5i15L9OC7rl7pwmYFR8QQgdM1ERWI",
+   						 	"alg": "EdDSA"
+						}`,
+			},
+		}
+
+		t.Parallel()
+
+		for _, test := range tests {
+			tc := test
+			t.Run(tc.name, func(t *testing.T) {
+				pk, err := DecodePublicKey([]byte(tc.jwkJSON))
+				require.NoError(t, err)
+				require.NotEmpty(t, pk)
+			})
+		}
+	})
+
+	t.Run("Test decode public key failure", func(t *testing.T) {
+		tests := []struct {
+			name    string
+			jwkJSON string
+			err     string
+		}{
+			{
+				name:    "attempt public key bytes from invalid JSON bytes",
+				jwkJSON: `}`,
+				err:     "unable to read JWK",
+			},
+			{
+				name: "attempt public key bytes from invalid curve",
+				jwkJSON: `{
+							"kty": "EC",
+    						"use": "enc",
+    						"crv": "sec12341",
+    						"kid": "sample@sample.id",
+    						"x": "wQehEGTVCu32yp8IwTaBCqPUIYslyd-WoFRsfDKE9II",
+    						"y": "rIJO8RmkExUecJ5i15L9OC7rl7pwmYFR8QQgdM1ERWI",
+   						 	"alg": "EdDSA"
+						}`,
+				err: "unsupported elliptic curve 'sec12341'",
+			},
+			{
+				name: "attempt public key bytes from invalid JSON bytes",
+				jwkJSON: `{
+							"kty": "EC",
+    						"use": "enc",
+    						"crv": "secp256k1",
+    						"kid": "sample@sample.id",
+    						"x": "",
+    						"y": "",
+   						 	"alg": "EdDSA"
+						}`,
+				err: "invalid EC public key, wrong length for x",
+			},
+			{
+				name: "attempt public key bytes from invalid JSON bytes",
+				jwkJSON: `{
+							"kty": "EC",
+    						"use": "enc",
+    						"crv": "secp256k1",
+    						"kid": "sample@sample.id",
+    						"x": "wQehEGTVCu32yp8IwTaBCqPUIYslyd-WoFRsfDKE9II",
+    						"y": "",
+   						 	"alg": "EdDSA"
```

Comment:
same here

Reply:
I see, alg is ignored by JWK parser since it uses 'kty=EC' and logic based on curve, I fixed tests by regenerating JWK using ES256(ECDSA using P-256)